### PR TITLE
Advanced Ballistics - Trajectory refinement

### PR DESCRIPTION
**Pull request by ulteq**
More advanced drag update routine:
* Handles greater `simulationIntervals` much better:
* `simulationInterval=0.00` -> Error (at 2km): 0.004 MILs
* `simulationInterval=0.05` -> Error (at 2km): 0.030 MILs
* `simulationInterval=0.10` -> Error (at 2km): 0.070 MILs
* Almost completely FPS invariant

Also:
* Disables BC degradation in the transonic zone (does not do justice to reality)
* Corrects the gravity constant (`9.80665` -> `9.8066`) in all modules
* Improves the random seed generator (used in the transonic flight calculations)
* Improves accuracy of the zero angle calculation (at greater zero distances)

- [x] Recompile DLLs

### CODE REVIEW

File: extensions/advanced_ballistics/AdvancedBallistics.cpp [hunk 1]
```diff
@@ -529,82 +524,123 @@ void __stdcall RVExtension(char *output, int outputSize, const char *function)
             windSpeed = sqrt(pow(wind[0], 2) + pow(wind[1], 2) + pow(wind[2], 2));
         }
 
-        trueVelocity[0] = bulletDatabase[index].bulletVelocity[0] - wind[0];
-        trueVelocity[1] = bulletDatabase[index].bulletVelocity[1] - wind[1];
-        trueVelocity[2] = bulletDatabase[index].bulletVelocity[2] - wind[2];
-        trueSpeed = sqrt(pow(trueVelocity[0], 2) + pow(trueVelocity[1], 2) + pow(trueVelocity[2], 2));
+        std::vector<double> bulletVelocity = bulletDatabase[index].bulletVelocity;
+        double time = 0.0f;
```
Comment (TheMagnetar) [inline extensions/advanced_ballistics/AdvancedBallistics.cpp]: double with float assignment

File: extensions/advanced_ballistics/AdvancedBallistics.cpp [hunk 2]
```diff
@@ -529,82 +524,123 @@ void __stdcall RVExtension(char *output, int outputSize, const char *function)
...
+        }
 
-        if (bulletDatabase[index].ballisticCoefficients.size() == bulletDatabase[index].velocityBoundaries.size() + 1) {
-            ballisticCoefficient = bulletDatabase[index].ballisticCoefficients[0];
-            for (int i = (int)bulletDatabase[index].velocityBoundaries.size() - 1; i >= 0; i = i - 1) {
-                if (trueSpeed < bulletDatabase[index].velocityBoundaries[i]) {
-                    ballisticCoefficient = bulletDatabase[index].ballisticCoefficients[i + 1];
-                    break;
+        bulletVelocity = bulletDatabase[index].bulletVelocity;
+        time = 0.0f;
```
Comment (TheMagnetar) [inline extensions/advanced_ballistics/AdvancedBallistics.cpp]: same as above

File: addons/advanced_ballistics/functions/fnc_handleFired.sqf [hunk 3]
```diff
@@ -112,7 +112,9 @@ if (_caliber > 0 && _bulletLength > 0 && _bulletMass > 0 && _barrelTwist > 0) th
 
 GVAR(currentbulletID) = (GVAR(currentbulletID) + 1) % 10000;
 
-"ace_advanced_ballistics" callExtension format["new:%1:%2:%3:%4:%5:%6:%7:%8:%9:%10:%11:%12:%13:%14:%15:%16:%17", GVAR(currentbulletID), _airFriction, _ballisticCoefficients, _velocityBoundaries, _atmosphereModel, _dragModel, _stabilityFactor, _twistDirection, _transonicStabilityCoef, getPosASL _projectile, _bulletVelocity, EGVAR(common,mapLatitude), EGVAR(weather,currentTemperature), EGVAR(common,mapAltitude), EGVAR(weather,currentHumidity), EGVAR(weather,currentOvercast), CBA_missionTime toFixed 6];
+private _ammoCount = _unit ammo (primaryWeapon _unit);
```
Comment (PabstMirror) [inline addons/advanced_ballistics/functions/fnc_handleFired.sqf]: I think AB runs for pistol shots (and maybe spotting rounds?), 
unit may not have a rifle
should be able to use `_weapon` or `_muzzle` from fired EH

Comment (ulteq) [reply addons/advanced_ballistics/functions/fnc_handleFired.sqf]: Yes, I should have used `_muzzle`. I will change it.

**Pull request merged**